ghga_connector/core/file_operations.py

```diff
@@ -18,18 +18,28 @@
 Contains Calls of the Presigned URLs in order to Up- and Download Files
 """
 
+from io import BytesIO
+
 import pycurl
-from ghga_service_chassis_lib.s3 import PresignedPostURL
 
 from .exceptions import BadResponseCodeError, RequestFailedError
 
 
 def download_file_part(
-    download_url: str, output_file_path: str, part_offset: int, part_end: int
-) -> None:  # pylint: disable=unused-argument
+    download_url: str,
+    output_file_path: str,
+    part_offset: int,
+    part_size: int,
+    file_size: int,
+) -> None:
     """Download File"""
 
-    with open(output_file_path, "ab") as file:
+    # Calculetes end of byte range of the file
+    part_end = part_offset + part_size - 1
+    if part_end > file_size:
+        part_end = file_size - 1
+
+    with open(file=output_file_path, mode="ab", buffering=part_size) as file:
```

Comment:
Does this actually work for the first file part?
i.e. opening in append mode should cause issues if the file does not yet exist at the target location
Please check if this works correctly

Reply:
Opening a non-existing file always creates that new file if it does not exist, no matter the mode.
Also, if it wouldn't work, the tests would fail.